cmd/kots/cli/download.go

```diff
@@ -34,21 +43,46 @@ func DownloadCmd() *cobra.Command {
 				}
 			}
 
+			output := v.GetString("output")
+			if output != "json" && output != "" {
+				return errors.Errorf("output format %s not supported (allowed formats are: json)", output)
+			}
+
 			downloadOptions := download.DownloadOptions{
 				Namespace:             v.GetString("namespace"),
 				Overwrite:             v.GetBool("overwrite"),
+				Silent:                output != "",
 				DecryptPasswordValues: v.GetBool("decrypt-password-values"),
 			}
 
+			var downloadOutput DownloadOutput
 			downloadPath := filepath.Join(ExpandDir(v.GetString("dest")), appSlug)
-			if err := download.Download(appSlug, downloadPath, downloadOptions); err != nil {
+			err := download.Download(appSlug, downloadPath, downloadOptions)
+			if err != nil && output == "" {
 				return errors.Cause(err)
+			} else if err != nil {
+				downloadOutput.Error = fmt.Sprint(errors.Cause(err))
+			} else {
+				downloadOutput.Success = true
+				downloadOutput.DownloadLocation = downloadPath
+				downloadOutput.UploadCommand = fmt.Sprintf("kubectl kots upload --namespace %s --slug %s %s", v.GetString("namespace"), appSlug, downloadPath)
 			}
 
 			log := logger.NewCLILogger()
+			if output == "json" {
+				outputJSON, err := json.Marshal(downloadOutput)
+				if err != nil {
+					return errors.Wrap(err, "error marshaling JSON")
+				}
+				log.Info(string(outputJSON))
```

Comment:
i would recommend adding a `return` after this?

Reply:
yeah that would remove the need to silence the logger again so i agree with that, thanks!